### PR TITLE
[PHX-845] Update QuestionBox to V5

### PR DESCRIPTION
**Pull request by micahhahn**
This PR creates `QuestionBox.V5`, which incorporates design changes planned for the scaffolding project. 

### Changes from `QuestionBox.V4`
- QuestionBox "themes" have been introduced.  We have 4: Neutral (blue), Correct (green), Incorrect (purple), and Tip (white). 
- The character icon **only** appears for the `Tip` theme.  The other themes are intended to be used in conjunction with the (upcoming) `CharacterStage` component. 
- The main box is no longer a `Balloon` component as it is too complex. 
- Buttons can now be styled as `primary` or `secondary`
- Buttons can be aligned vertically or horizontally
- All QuestionBox positioning logic has been dropped as we no longer need it 
- The `leftActions` API interaction is drastically improved
  - The QuestionBox now takes the responsibility of centering the contents by their width in the top left part of the box

[Here are the new designs on Figma for reference](https://www.figma.com/file/swxmvrKsgawbuicqHtdwN8/Learning-XP-(2023%2B)?type=design&node-id=572-10889&t=UGuqxXD6Cbh5hJAF-0)

# :wrench: Modifying a component

## Context

Fix https://linear.app/noredink/issue/PHX-845/questionboxv5-component-upgrade

## :framed_picture: What does this change look like?

https://github.com/NoRedInk/noredink-ui/assets/12899207/d97366ec-6244-4926-b067-26c1ccdae4d7

## Component completion checklist

- [x] I've gone through the relevant sections of the [Development Accessibility guide](https://paper.dropbox.com/doc/Accessibility-guide-4-Development--BiIVdijSaoijjOuhz3iTCJJ1Ag-rGoHpC91pFg3zTrYpvOCQ) with the changes I made to this component in mind
- [x] Changes are clearly documented
    - [x] Component docs include a changelog
    - [x] Any new exposed functions or properties have docs
- [x] Changes extend to the Component Catalog
    - [x] The Component Catalog is updated to use the newest version, if appropriate
    - [x] The Component Catalog example version number is updated, if appropriate
    - [x] Any new customizations are available from the Component Catalog
    - [x] The component example still has:
        - an accurate preview
        - valid sample code
        - correct keyboard behavior
        - correct and comprehensive guidance around how to use the component
- [x] Changes to the component are tested/the new version of the component is tested
- [x] Component API follows standard patterns in noredink-ui
    - e.g., as a dev, I can conveniently add an `nriDescription`
    - and adding a new feature to the component will _not_ require major API changes to the component

## :microscope: What do you want the reviewer to check?

I put in a little extra effort to get the generated Elm codes to be correct again so that the Ellie link works. 

I've also tried to have all the different design configurations of the QuestionBox be represented in the catalog examples.

### CODE REVIEW

**Comment by micahhahn** [comment]
> Can we infer the button theme from how many actions there are and which QB theme is being used?

Unfortunately not! The general case will be primary if there is only 1 button and secondary for multiple button.  But there are several exceptions in the design where we want to highlight one button over the others so I thought it was best to make the API powerful enough to handle that. 

> Will the horizontal/vertical alignment of the buttons be consistent for the various themes?

No not really either.   There are only a couple exceptions where we will want to have buttons be horizontal so I think we need to allow setting it from the API. (Design has an idea to default to being horizontal unless the buttons wouldn't fit in one row, but that has not materialized yet).

> Also, I think we need to ensure that buttons can reflow so that they don't end up rendering outside the QB for mobile:

Agreed! FWIW, we will only have horizontal buttons for cases where we know **exactly** what the button contents will be - so there isn't risk of overflow really.  But I'll tweak the flexbox settings just to be safe.

File: component-catalog/src/Examples/QuestionBox.elm [hunk 1]
```diff
@@ -302,35 +327,24 @@ initAttributes =
...
-                        )                      """
+                    ( Code.fromModule moduleName "leftActions" ++ Code.newlineWithIndent 2 ++ leftActionIconsCode [ ( "Play", "playInCircle" ) ]
                     , QuestionBox.leftActions
-                        ([ ClickableSvg.button "Play"
-                            UiIcon.playInCircle
-                            [ ClickableSvg.exactSize 24
-                            , ClickableSvg.css
-                                [ Css.borderRadius (Css.px 24)
-                                , Css.backgroundColor Colors.white
+                        (div []
```
[the remark this answers is not in the document]
Comment (micahhahn) [reply component-catalog/src/Examples/QuestionBox.elm:337]: No, but it does allow me to re-use `leftActionIconsCode`.

File: src/Nri/Ui/QuestionBox/V5.elm [hunk 2]
```diff
@@ -47,14 +47,18 @@ type Attribute msg
...
+    , actions : List (Action msg)
     , theme : QuestionBoxTheme
     , character : Maybe { name : String, icon : Svg }
     , containerCss : List Css.Style
     , leftActions : Maybe (Html msg)
     }
 
 
+type alias Action msg =
+    { label : String, theme : Button.Attribute msg, onClick : msg }
```
[the remark this answers is not in the document]
Comment (micahhahn) [reply src/Nri/Ui/QuestionBox/V5.elm:56]: I definitely thought about this.  One thing to keep in mind is that we are trying to have the API manage some things automatically such as the button size (`medium` vs `small` for one button or many buttons) and button text alignment.  I didn't really want to give the illusion of full control of the button - but I also didn't make it a boolean or something either. 

Do you think changing it to something like `isPrimary : Bool` would be better?

**Comment by tesk9** [comment]
Thanks for the updates! This is looking great. I'm a little concerned about CI not working, so I've asked Brian to take a look. If this is time sensitive, LMK, and I'll approve instead and we can move forward and deal with CI separately.

**Comment by tesk9** [comment]
(Brian said merging master should fix CI, so giving that a try!)

**Review by tesk9 — APPROVED** [review]

**Pull request merged**